cmd/extproc/mainlib/main.go

```diff
@@ -120,6 +131,28 @@ func Main() {
 		<-ctx.Done()
 		s.GracefulStop()
 	}()
+
+	// Add prometheus metrics
+	metricsInstance := metrics.GetOrCreate()
+
+	// Setup prometheus handler
+	http.Handle("/metrics", promhttp.HandlerFor(metricsInstance.Registry, promhttp.HandlerOpts{}))
+
+	go func() {
+		server := &http.Server{
```

Comment:
nit: You may want to define the `server` outside this goroutine so you can also call `server.Shutdown`, in the goroutine above that waits for the context to be done, this way we gracefully terminate everything.

Reply:
updated